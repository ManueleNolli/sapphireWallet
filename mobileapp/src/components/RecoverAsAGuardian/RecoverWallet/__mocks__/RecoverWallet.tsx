const RecoverWalletMock = () => <div data-testid="RecoverWallet">RecoverWallet</div>

export default RecoverWalletMock
