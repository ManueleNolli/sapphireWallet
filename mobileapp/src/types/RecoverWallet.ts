export type RecoverWallet = {
  walletAddress: string
  chainID: string
  nonce: string
  wrappedTransaction: string
  signedTransaction: string
  mnemonic: string[]
}

export const RecoverWalletToJSON = (value: RecoverWallet): string => {
  const mnemonic = value.mnemonic.join(' ')
  return JSON.stringify({ ...value, mnemonic })
}

export const JSONToRecoverWallet = (value: string): RecoverWallet => {
  const parsed = JSON.parse(value)
  return { ...parsed, mnemonic: parsed.mnemonic.split(' ') }
}
