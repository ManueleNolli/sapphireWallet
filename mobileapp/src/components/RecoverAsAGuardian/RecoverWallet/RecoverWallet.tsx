import { Button, Layout, Spinner, StyleService, Text } from '@ui-kitten/components'
import { appStyles, vh, vw } from '../../../Styles'
import React from 'react'
import useRecoverWallet from './useRecoverWallet'
import { View } from 'react-native'
import QRCode from 'react-native-qrcode-svg'

type RecoverWalletProps = {
  wallet: string
}

export default function RecoverWallet(props: RecoverWalletProps) {
  const { isLoading, prepareTransactionAndShowQrCode, recoverWalletInfo } = useRecoverWallet(props)

  return (
    <Layout style={styles.modalContainer}>
      <Text category="h6" style={{ marginBottom: 2 * vh }}>
        Recover wallet
      </Text>

      {recoverWalletInfo ? (
        <QRCode value={recoverWalletInfo} size={75 * vw} />
      ) : isLoading ? (
        <View style={appStyles.center}>
          <Spinner
            size="giant"
            status="basic"
            style={{
              marginBottom: 2 * vh,
            }}
          />
        </View>
      ) : (
        <>
          <Text style={{ textAlign: 'center' }}>Are you sure you want to recover wallet</Text>
          <Text
            style={{
              textAlign: 'center',
              marginBottom: 2 * vh,
            }}
            category="label"
          >
            {props.wallet}
          </Text>

          <Button testID="button-yen" style={{ width: '50%' }} onPress={prepareTransactionAndShowQrCode}>
            Yes
          </Button>
        </>
      )}
    </Layout>
  )
}

const styles = StyleService.create({
  modalContainer: {
    maxHeight: 75 * vh,
    minHeight: 25 * vh,
    maxWidth: 90 * vw,
    minWidth: 90 * vw,
    borderRadius: 10,
    paddingVertical: 4 * vh,
    paddingHorizontal: 4 * vw,
    alignItems: 'center',
  },
})
