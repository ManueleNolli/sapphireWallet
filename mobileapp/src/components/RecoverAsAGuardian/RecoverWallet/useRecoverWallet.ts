import { useContext, useState } from 'react'
import useLoading from '../../../hooks/useLoading'
import { RecoverWallet, RecoverWalletToJSON } from '../../../types/RecoverWallet'
import { prepareRecoverWallet } from '../../../services/transactions'
import { createWallet, getSigner } from '../../../services/wallet'
import { WalletContext } from '../../../context/WalletContext'
import { BlockchainContext } from '../../../context/BlockchainContext'

type useRecoverWalletProps = {
  wallet: string
}

export function useRecoverWallet({ wallet }: useRecoverWalletProps) {
  const { getPrivateKey } = useContext(WalletContext)
  const { currentNetwork } = useContext(BlockchainContext)
  const { isLoading, setIsLoading } = useLoading()
  const [recoverWalletInfo, setRecoverWalletInfo] = useState<string | undefined>()

  const prepareTransactionAndShowQrCode = async () => {
    setIsLoading(true)
    const signer = await getSigner(await getPrivateKey('Confirm recovery by signing the transaction'), currentNetwork)
    const newWallet = createWallet()
    const data = await prepareRecoverWallet(signer, currentNetwork, wallet, newWallet)
    const dataString = RecoverWalletToJSON(data)
    setRecoverWalletInfo(dataString)
    setIsLoading(false)
  }

  return {
    isLoading,
    prepareTransactionAndShowQrCode,
    recoverWalletInfo,
  }
}
