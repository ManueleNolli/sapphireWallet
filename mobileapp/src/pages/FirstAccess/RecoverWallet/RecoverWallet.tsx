import React from 'react'
import { Button, Layout, Modal, Spinner, Text } from '@ui-kitten/components'
import { RecoverWalletProps } from '../../../navigation/FirstAccessStack'
import SafeAreaView from '../../../utils/SafeAreaView'
import { appStyles, vh } from '../../../Styles'
import QRCodeScanner from '../../../components/QRCodeScanner/QRCodeScanner'
import { useRecoverWallet } from './useRecoverWallet'
import { View } from 'react-native'

export default function RecoverWallet({ route, navigation }: RecoverWalletProps) {
  const { isScannedOpen, closeModal, openModal, onQrCodeScanned, isLoading } = useRecoverWallet({ route, navigation })
  return (
    <Layout style={{ flex: 1 }}>
      <SafeAreaView style={{ flex: 1, justifyContent: 'center' }}>
        <Text category="h4" style={{ marginBottom: 2 * vh, textAlign: 'center' }}>
          Recover wallet
        </Text>
        <Text style={{ textAlign: 'center' }}>Scan the QR code of your Guardian to recover your wallet</Text>

        {isLoading ? (
          <View style={appStyles.center}>
            <Spinner
              size="giant"
              status="basic"
              style={{
                marginBottom: 2 * vh,
              }}
            />
          </View>
        ) : (
          <Button appearance="outline" size="giant" onPress={openModal} style={{ marginTop: 6 * vh }}>
            Scan QR code
          </Button>
        )}

        <Modal
          visible={isScannedOpen}
          backdropStyle={{ backgroundColor: 'color-basic-transparent-600' }}
          onBackdropPress={closeModal}
        >
          <QRCodeScanner onQRCodeScanned={onQrCodeScanned} />
        </Modal>
      </SafeAreaView>
    </Layout>
  )
}
