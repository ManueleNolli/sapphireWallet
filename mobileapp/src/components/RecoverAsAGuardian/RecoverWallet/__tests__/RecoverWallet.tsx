import useRecoverWallet from '../useRecoverWallet'
import renderWithTheme from '../../../../TestHelper'
import React from 'react'
import RecoverWallet from '../RecoverWallet'

jest.mock('../useRecoverWallet', () => jest.fn())
jest.mock('@ui-kitten/components', () => {
  const { View } = require('react-native')
  return {
    ...jest.requireActual('@ui-kitten/components'),
    Spinner: () => <View data-testid="spinner" />,
  }
})
describe('RecoverWallet', () => {
  it('should render correctly the button', () => {
    ;(useRecoverWallet as jest.Mock).mockReturnValue({
      isLoading: false,
      prepareTransactionAndShowQrCode: jest.fn(),
      recoverWalletInfo: '',
    })

    const tree = renderWithTheme(<RecoverWallet wallet="wallet" />)

    expect(tree.getByTestId('button-yen')).toBeTruthy()

    expect(tree).toMatchSnapshot()
  })

  it('should render correctly the qr code', () => {
    ;(useRecoverWallet as jest.Mock).mockReturnValue({
      isLoading: false,
      prepareTransactionAndShowQrCode: jest.fn(),
      recoverWalletInfo: 'walletInfo',
    })

    const tree = renderWithTheme(<RecoverWallet wallet="wallet" />)

    expect(tree).toMatchSnapshot()
  })

  it('should render correctly when loading', () => {
    ;(useRecoverWallet as jest.Mock).mockReturnValue({
      isLoading: true,
      prepareTransactionAndShowQrCode: jest.fn(),
      recoverWalletInfo: 'walletInfo',
    })

    const tree = renderWithTheme(<RecoverWallet wallet="wallet" />)

    expect(tree).toMatchSnapshot()
  })
})
