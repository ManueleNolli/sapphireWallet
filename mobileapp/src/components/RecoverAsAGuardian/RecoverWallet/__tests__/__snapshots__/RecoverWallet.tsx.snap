// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecoverWallet should render correctly the button 1`] = `
<View
  appearance="default"
  level="1"
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
      },
      {
        "alignItems": "center",
        "borderRadius": 10,
        "maxHeight": 1000.5,
        "maxWidth": 675,
        "minHeight": 333.5,
        "minWidth": 675,
        "paddingHorizontal": 30,
        "paddingVertical": 53.36,
      },
    ]
  }
>
  <Text
    appearance="default"
    category="h6"
    status="basic"
    style={
      [
        {
          "color": "#222B45",
          "fontFamily": "System",
          "fontSize": 18,
          "fontWeight": "800",
        },
        {
          "marginBottom": 26.68,
        },
      ]
    }
  >
    Recover wallet
  </Text>
  <Text
    appearance="default"
    category="p1"
    status="basic"
    style={
      [
        {
          "color": "#222B45",
          "fontFamily": "System",
          "fontSize": 15,
          "fontWeight": "400",
        },
        {
          "textAlign": "center",
        },
      ]
    }
  >
    Are you sure you want to recover wallet
  </Text>
  <Text
    appearance="default"
    category="label"
    status="basic"
    style={
      [
        {
          "color": "#222B45",
          "fontFamily": "System",
          "fontSize": 12,
          "fontWeight": "800",
        },
        {
          "marginBottom": 26.68,
          "textAlign": "center",
        },
      ]
    }
  >
    wallet
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#3366FF",
        "borderColor": "#3366FF",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "minHeight": 40,
        "minWidth": 40,
        "opacity": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 12,
        "width": "50%",
      }
    }
    testID="button-yen"
  >
    <Text
      appearance="default"
      category="p1"
      status="basic"
      style={
        [
          {
            "color": "#222B45",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
          },
          {
            "color": "#FFFFFF",
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "bold",
            "marginHorizontal": 10,
          },
        ]
      }
    >
      Yes
    </Text>
  </View>
</View>
`;

exports[`RecoverWallet should render correctly the qr code 1`] = `
<View
  appearance="default"
  level="1"
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
      },
      {
        "alignItems": "center",
        "borderRadius": 10,
        "maxHeight": 1000.5,
        "maxWidth": 675,
        "minHeight": 333.5,
        "minWidth": 675,
        "paddingHorizontal": 30,
        "paddingVertical": 53.36,
      },
    ]
  }
>
  <Text
    appearance="default"
    category="h6"
    status="basic"
    style={
      [
        {
          "color": "#222B45",
          "fontFamily": "System",
          "fontSize": 18,
          "fontWeight": "800",
        },
        {
          "marginBottom": 26.68,
        },
      ]
    }
  >
    Recover wallet
  </Text>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="562.5"
    bbWidth="562.5"
    focusable={false}
    height={562.5}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 562,
          "width": 562,
        },
      ]
    }
    vbHeight={562.5}
    vbWidth={562.5}
    width={562.5}
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
    >
      <RNSVGDefs>
        <RNSVGLinearGradient
          gradient={
            [
              0,
              -65536,
              1,
              -16711681,
            ]
          }
          gradientTransform={null}
          gradientUnits={0}
          name="grad"
          x1="0%"
          x2="100%"
          y1="0%"
          y2="100%"
        />
      </RNSVGDefs>
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGRect
          fill={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          height="562.5"
          propList={
            [
              "fill",
            ]
          }
          width="562.5"
          x="0"
          y="0"
        />
      </RNSVGGroup>
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGPath
          d="M0 13.392857142857142 L187.5 13.392857142857142 M241.07142857142856 13.392857142857142 L267.85714285714283 13.392857142857142 M375 13.392857142857142 L562.5 13.392857142857142 M0 40.17857142857143 L26.785714285714285 40.17857142857143 M160.71428571428572 40.17857142857143 L187.5 40.17857142857143 M267.85714285714283 40.17857142857143 L321.42857142857144 40.17857142857143 M375 40.17857142857143 L401.7857142857143 40.17857142857143 M535.7142857142857 40.17857142857143 L562.5 40.17857142857143 M0 66.96428571428571 L26.785714285714285 66.96428571428571 M53.57142857142857 66.96428571428571 L133.92857142857142 66.96428571428571 M160.71428571428572 66.96428571428571 L187.5 66.96428571428571 M214.28571428571428 66.96428571428571 L241.07142857142856 66.96428571428571 M267.85714285714283 66.96428571428571 L294.6428571428571 66.96428571428571 M321.42857142857144 66.96428571428571 L348.2142857142857 66.96428571428571 M375 66.96428571428571 L401.7857142857143 66.96428571428571 M428.57142857142856 66.96428571428571 L508.9285714285714 66.96428571428571 M535.7142857142857 66.96428571428571 L562.5 66.96428571428571 M0 93.75 L26.785714285714285 93.75 M53.57142857142857 93.75 L133.92857142857142 93.75 M160.71428571428572 93.75 L187.5 93.75 M214.28571428571428 93.75 L241.07142857142856 93.75 M321.42857142857144 93.75 L348.2142857142857 93.75 M375 93.75 L401.7857142857143 93.75 M428.57142857142856 93.75 L508.9285714285714 93.75 M535.7142857142857 93.75 L562.5 93.75 M0 120.53571428571428 L26.785714285714285 120.53571428571428 M53.57142857142857 120.53571428571428 L133.92857142857142 120.53571428571428 M160.71428571428572 120.53571428571428 L187.5 120.53571428571428 M214.28571428571428 120.53571428571428 L241.07142857142856 120.53571428571428 M267.85714285714283 120.53571428571428 L348.2142857142857 120.53571428571428 M375 120.53571428571428 L401.7857142857143 120.53571428571428 M428.57142857142856 120.53571428571428 L508.9285714285714 120.53571428571428 M535.7142857142857 120.53571428571428 L562.5 120.53571428571428 M0 147.32142857142856 L26.785714285714285 147.32142857142856 M160.71428571428572 147.32142857142856 L187.5 147.32142857142856 M214.28571428571428 147.32142857142856 L241.07142857142856 147.32142857142856 M267.85714285714283 147.32142857142856 L294.6428571428571 147.32142857142856 M321.42857142857144 147.32142857142856 L348.2142857142857 147.32142857142856 M375 147.32142857142856 L401.7857142857143 147.32142857142856 M535.7142857142857 147.32142857142856 L562.5 147.32142857142856 M0 174.10714285714286 L187.5 174.10714285714286 M214.28571428571428 174.10714285714286 L241.07142857142856 174.10714285714286 M267.85714285714283 174.10714285714286 L294.6428571428571 174.10714285714286 M321.42857142857144 174.10714285714286 L348.2142857142857 174.10714285714286 M375 174.10714285714286 L562.5 174.10714285714286 M214.28571428571428 200.89285714285714 L348.2142857142857 200.89285714285714 M0 227.67857142857142 L26.785714285714285 227.67857142857142 M53.57142857142857 227.67857142857142 L187.5 227.67857142857142 M241.07142857142856 227.67857142857142 L267.85714285714283 227.67857142857142 M321.42857142857144 227.67857142857142 L348.2142857142857 227.67857142857142 M375 227.67857142857142 L508.9285714285714 227.67857142857142 M0 254.4642857142857 L26.785714285714285 254.4642857142857 M53.57142857142857 254.4642857142857 L107.14285714285714 254.4642857142857 M214.28571428571428 254.4642857142857 L267.85714285714283 254.4642857142857 M321.42857142857144 254.4642857142857 L455.35714285714283 254.4642857142857 M482.1428571428571 254.4642857142857 L508.9285714285714 254.4642857142857 M535.7142857142857 254.4642857142857 L562.5 254.4642857142857 M0 281.25 L107.14285714285714 281.25 M133.92857142857142 281.25 L214.28571428571428 281.25 M241.07142857142856 281.25 L267.85714285714283 281.25 M294.6428571428571 281.25 L321.42857142857144 281.25 M375 281.25 L401.7857142857143 281.25 M455.35714285714283 281.25 L535.7142857142857 281.25 M80.35714285714286 308.0357142857143 L160.71428571428572 308.0357142857143 M187.5 308.0357142857143 L267.85714285714283 308.0357142857143 M348.2142857142857 308.0357142857143 L375 308.0357142857143 M455.35714285714283 308.0357142857143 L562.5 308.0357142857143 M26.785714285714285 334.8214285714286 L53.57142857142857 334.8214285714286 M80.35714285714286 334.8214285714286 L107.14285714285714 334.8214285714286 M133.92857142857142 334.8214285714286 L187.5 334.8214285714286 M241.07142857142856 334.8214285714286 L267.85714285714283 334.8214285714286 M294.6428571428571 334.8214285714286 L321.42857142857144 334.8214285714286 M375 334.8214285714286 L455.35714285714283 334.8214285714286 M508.9285714285714 334.8214285714286 L535.7142857142857 334.8214285714286 M214.28571428571428 361.6071428571429 L241.07142857142856 361.6071428571429 M267.85714285714283 361.6071428571429 L375 361.6071428571429 M401.7857142857143 361.6071428571429 L428.57142857142856 361.6071428571429 M455.35714285714283 361.6071428571429 L482.1428571428571 361.6071428571429 M508.9285714285714 361.6071428571429 L562.5 361.6071428571429 M0 388.39285714285717 L187.5 388.39285714285717 M241.07142857142856 388.39285714285717 L294.6428571428571 388.39285714285717 M321.42857142857144 388.39285714285717 L348.2142857142857 388.39285714285717 M375 388.39285714285717 L401.7857142857143 388.39285714285717 M482.1428571428571 388.39285714285717 L535.7142857142857 388.39285714285717 M0 415.17857142857144 L26.785714285714285 415.17857142857144 M160.71428571428572 415.17857142857144 L187.5 415.17857142857144 M214.28571428571428 415.17857142857144 L241.07142857142856 415.17857142857144 M267.85714285714283 415.17857142857144 L375 415.17857142857144 M401.7857142857143 415.17857142857144 L428.57142857142856 415.17857142857144 M455.35714285714283 415.17857142857144 L508.9285714285714 415.17857142857144 M535.7142857142857 415.17857142857144 L562.5 415.17857142857144 M0 441.9642857142857 L26.785714285714285 441.9642857142857 M53.57142857142857 441.9642857142857 L133.92857142857142 441.9642857142857 M160.71428571428572 441.9642857142857 L187.5 441.9642857142857 M214.28571428571428 441.9642857142857 L294.6428571428571 441.9642857142857 M321.42857142857144 441.9642857142857 L348.2142857142857 441.9642857142857 M375 441.9642857142857 L401.7857142857143 441.9642857142857 M535.7142857142857 441.9642857142857 L562.5 441.9642857142857 M0 468.75 L26.785714285714285 468.75 M53.57142857142857 468.75 L133.92857142857142 468.75 M160.71428571428572 468.75 L187.5 468.75 M214.28571428571428 468.75 L241.07142857142856 468.75 M321.42857142857144 468.75 L401.7857142857143 468.75 M428.57142857142856 468.75 L482.1428571428571 468.75 M0 495.5357142857143 L26.785714285714285 495.5357142857143 M53.57142857142857 495.5357142857143 L133.92857142857142 495.5357142857143 M160.71428571428572 495.5357142857143 L187.5 495.5357142857143 M214.28571428571428 495.5357142857143 L321.42857142857144 495.5357142857143 M348.2142857142857 495.5357142857143 L401.7857142857143 495.5357142857143 M482.1428571428571 495.5357142857143 L508.9285714285714 495.5357142857143 M0 522.3214285714286 L26.785714285714285 522.3214285714286 M160.71428571428572 522.3214285714286 L187.5 522.3214285714286 M267.85714285714283 522.3214285714286 L294.6428571428571 522.3214285714286 M455.35714285714283 522.3214285714286 L508.9285714285714 522.3214285714286 M0 549.1071428571428 L187.5 549.1071428571428 M214.28571428571428 549.1071428571428 L241.07142857142856 549.1071428571428 M294.6428571428571 549.1071428571428 L321.42857142857144 549.1071428571428 M508.9285714285714 549.1071428571428 L535.7142857142857 549.1071428571428 "
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          propList={
            [
              "stroke",
              "strokeWidth",
              "strokeLinecap",
            ]
          }
          stroke={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          strokeLinecap={0}
          strokeWidth="26.785714285714285"
        />
      </RNSVGGroup>
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`RecoverWallet should render correctly when loading 1`] = `
<View
  appearance="default"
  level="1"
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
      },
      {
        "alignItems": "center",
        "borderRadius": 10,
        "maxHeight": 1000.5,
        "maxWidth": 675,
        "minHeight": 333.5,
        "minWidth": 675,
        "paddingHorizontal": 30,
        "paddingVertical": 53.36,
      },
    ]
  }
>
  <Text
    appearance="default"
    category="h6"
    status="basic"
    style={
      [
        {
          "color": "#222B45",
          "fontFamily": "System",
          "fontSize": 18,
          "fontWeight": "800",
        },
        {
          "marginBottom": 26.68,
        },
      ]
    }
  >
    Recover wallet
  </Text>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="562.5"
    bbWidth="562.5"
    focusable={false}
    height={562.5}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 562,
          "width": 562,
        },
      ]
    }
    vbHeight={562.5}
    vbWidth={562.5}
    width={562.5}
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
    >
      <RNSVGDefs>
        <RNSVGLinearGradient
          gradient={
            [
              0,
              -65536,
              1,
              -16711681,
            ]
          }
          gradientTransform={null}
          gradientUnits={0}
          name="grad"
          x1="0%"
          x2="100%"
          y1="0%"
          y2="100%"
        />
      </RNSVGDefs>
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGRect
          fill={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          height="562.5"
          propList={
            [
              "fill",
            ]
          }
          width="562.5"
          x="0"
          y="0"
        />
      </RNSVGGroup>
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGPath
          d="M0 13.392857142857142 L187.5 13.392857142857142 M241.07142857142856 13.392857142857142 L267.85714285714283 13.392857142857142 M375 13.392857142857142 L562.5 13.392857142857142 M0 40.17857142857143 L26.785714285714285 40.17857142857143 M160.71428571428572 40.17857142857143 L187.5 40.17857142857143 M267.85714285714283 40.17857142857143 L321.42857142857144 40.17857142857143 M375 40.17857142857143 L401.7857142857143 40.17857142857143 M535.7142857142857 40.17857142857143 L562.5 40.17857142857143 M0 66.96428571428571 L26.785714285714285 66.96428571428571 M53.57142857142857 66.96428571428571 L133.92857142857142 66.96428571428571 M160.71428571428572 66.96428571428571 L187.5 66.96428571428571 M214.28571428571428 66.96428571428571 L241.07142857142856 66.96428571428571 M267.85714285714283 66.96428571428571 L294.6428571428571 66.96428571428571 M321.42857142857144 66.96428571428571 L348.2142857142857 66.96428571428571 M375 66.96428571428571 L401.7857142857143 66.96428571428571 M428.57142857142856 66.96428571428571 L508.9285714285714 66.96428571428571 M535.7142857142857 66.96428571428571 L562.5 66.96428571428571 M0 93.75 L26.785714285714285 93.75 M53.57142857142857 93.75 L133.92857142857142 93.75 M160.71428571428572 93.75 L187.5 93.75 M214.28571428571428 93.75 L241.07142857142856 93.75 M321.42857142857144 93.75 L348.2142857142857 93.75 M375 93.75 L401.7857142857143 93.75 M428.57142857142856 93.75 L508.9285714285714 93.75 M535.7142857142857 93.75 L562.5 93.75 M0 120.53571428571428 L26.785714285714285 120.53571428571428 M53.57142857142857 120.53571428571428 L133.92857142857142 120.53571428571428 M160.71428571428572 120.53571428571428 L187.5 120.53571428571428 M214.28571428571428 120.53571428571428 L241.07142857142856 120.53571428571428 M267.85714285714283 120.53571428571428 L348.2142857142857 120.53571428571428 M375 120.53571428571428 L401.7857142857143 120.53571428571428 M428.57142857142856 120.53571428571428 L508.9285714285714 120.53571428571428 M535.7142857142857 120.53571428571428 L562.5 120.53571428571428 M0 147.32142857142856 L26.785714285714285 147.32142857142856 M160.71428571428572 147.32142857142856 L187.5 147.32142857142856 M214.28571428571428 147.32142857142856 L241.07142857142856 147.32142857142856 M267.85714285714283 147.32142857142856 L294.6428571428571 147.32142857142856 M321.42857142857144 147.32142857142856 L348.2142857142857 147.32142857142856 M375 147.32142857142856 L401.7857142857143 147.32142857142856 M535.7142857142857 147.32142857142856 L562.5 147.32142857142856 M0 174.10714285714286 L187.5 174.10714285714286 M214.28571428571428 174.10714285714286 L241.07142857142856 174.10714285714286 M267.85714285714283 174.10714285714286 L294.6428571428571 174.10714285714286 M321.42857142857144 174.10714285714286 L348.2142857142857 174.10714285714286 M375 174.10714285714286 L562.5 174.10714285714286 M214.28571428571428 200.89285714285714 L348.2142857142857 200.89285714285714 M0 227.67857142857142 L26.785714285714285 227.67857142857142 M53.57142857142857 227.67857142857142 L187.5 227.67857142857142 M241.07142857142856 227.67857142857142 L267.85714285714283 227.67857142857142 M321.42857142857144 227.67857142857142 L348.2142857142857 227.67857142857142 M375 227.67857142857142 L508.9285714285714 227.67857142857142 M0 254.4642857142857 L26.785714285714285 254.4642857142857 M53.57142857142857 254.4642857142857 L107.14285714285714 254.4642857142857 M214.28571428571428 254.4642857142857 L267.85714285714283 254.4642857142857 M321.42857142857144 254.4642857142857 L455.35714285714283 254.4642857142857 M482.1428571428571 254.4642857142857 L508.9285714285714 254.4642857142857 M535.7142857142857 254.4642857142857 L562.5 254.4642857142857 M0 281.25 L107.14285714285714 281.25 M133.92857142857142 281.25 L214.28571428571428 281.25 M241.07142857142856 281.25 L267.85714285714283 281.25 M294.6428571428571 281.25 L321.42857142857144 281.25 M375 281.25 L401.7857142857143 281.25 M455.35714285714283 281.25 L535.7142857142857 281.25 M80.35714285714286 308.0357142857143 L160.71428571428572 308.0357142857143 M187.5 308.0357142857143 L267.85714285714283 308.0357142857143 M348.2142857142857 308.0357142857143 L375 308.0357142857143 M455.35714285714283 308.0357142857143 L562.5 308.0357142857143 M26.785714285714285 334.8214285714286 L53.57142857142857 334.8214285714286 M80.35714285714286 334.8214285714286 L107.14285714285714 334.8214285714286 M133.92857142857142 334.8214285714286 L187.5 334.8214285714286 M241.07142857142856 334.8214285714286 L267.85714285714283 334.8214285714286 M294.6428571428571 334.8214285714286 L321.42857142857144 334.8214285714286 M375 334.8214285714286 L455.35714285714283 334.8214285714286 M508.9285714285714 334.8214285714286 L535.7142857142857 334.8214285714286 M214.28571428571428 361.6071428571429 L241.07142857142856 361.6071428571429 M267.85714285714283 361.6071428571429 L375 361.6071428571429 M401.7857142857143 361.6071428571429 L428.57142857142856 361.6071428571429 M455.35714285714283 361.6071428571429 L482.1428571428571 361.6071428571429 M508.9285714285714 361.6071428571429 L562.5 361.6071428571429 M0 388.39285714285717 L187.5 388.39285714285717 M241.07142857142856 388.39285714285717 L294.6428571428571 388.39285714285717 M321.42857142857144 388.39285714285717 L348.2142857142857 388.39285714285717 M375 388.39285714285717 L401.7857142857143 388.39285714285717 M482.1428571428571 388.39285714285717 L535.7142857142857 388.39285714285717 M0 415.17857142857144 L26.785714285714285 415.17857142857144 M160.71428571428572 415.17857142857144 L187.5 415.17857142857144 M214.28571428571428 415.17857142857144 L241.07142857142856 415.17857142857144 M267.85714285714283 415.17857142857144 L375 415.17857142857144 M401.7857142857143 415.17857142857144 L428.57142857142856 415.17857142857144 M455.35714285714283 415.17857142857144 L508.9285714285714 415.17857142857144 M535.7142857142857 415.17857142857144 L562.5 415.17857142857144 M0 441.9642857142857 L26.785714285714285 441.9642857142857 M53.57142857142857 441.9642857142857 L133.92857142857142 441.9642857142857 M160.71428571428572 441.9642857142857 L187.5 441.9642857142857 M214.28571428571428 441.9642857142857 L294.6428571428571 441.9642857142857 M321.42857142857144 441.9642857142857 L348.2142857142857 441.9642857142857 M375 441.9642857142857 L401.7857142857143 441.9642857142857 M535.7142857142857 441.9642857142857 L562.5 441.9642857142857 M0 468.75 L26.785714285714285 468.75 M53.57142857142857 468.75 L133.92857142857142 468.75 M160.71428571428572 468.75 L187.5 468.75 M214.28571428571428 468.75 L241.07142857142856 468.75 M321.42857142857144 468.75 L401.7857142857143 468.75 M428.57142857142856 468.75 L482.1428571428571 468.75 M0 495.5357142857143 L26.785714285714285 495.5357142857143 M53.57142857142857 495.5357142857143 L133.92857142857142 495.5357142857143 M160.71428571428572 495.5357142857143 L187.5 495.5357142857143 M214.28571428571428 495.5357142857143 L321.42857142857144 495.5357142857143 M348.2142857142857 495.5357142857143 L401.7857142857143 495.5357142857143 M482.1428571428571 495.5357142857143 L508.9285714285714 495.5357142857143 M0 522.3214285714286 L26.785714285714285 522.3214285714286 M160.71428571428572 522.3214285714286 L187.5 522.3214285714286 M267.85714285714283 522.3214285714286 L294.6428571428571 522.3214285714286 M455.35714285714283 522.3214285714286 L508.9285714285714 522.3214285714286 M0 549.1071428571428 L187.5 549.1071428571428 M214.28571428571428 549.1071428571428 L241.07142857142856 549.1071428571428 M294.6428571428571 549.1071428571428 L321.42857142857144 549.1071428571428 M508.9285714285714 549.1071428571428 L535.7142857142857 549.1071428571428 "
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          propList={
            [
              "stroke",
              "strokeWidth",
              "strokeLinecap",
            ]
          }
          stroke={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          strokeLinecap={0}
          strokeWidth="26.785714285714285"
        />
      </RNSVGGroup>
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
